articles/container-components-in-react/index.md

```diff
@@ -0,0 +1,178 @@
+
+If you look at the component above, you’d see that it makes use of props to share data among components in the “arbitrary” React app.
+
+**Note:** There’s a convention that is quite common in the developer community.
+
+Developers tend to place their container components in a separate folder and the UI components into another folder, so they can distinguish between these components. You can choose not to follow this pattern of creating different folders if you already have one that works fine for you, so long you’re able to solve the same problem.
+
+### Building the search component
+
+A search component is among the most ubiquitous components in a typical React application.
+
+Let’s start by thinking about the function(s) that the search component would perform. A search component should be a able to perform basic tasks like searching and filtering.
+
+To keep things simple, let's use an array of objects as the data to search from, instead of using an API.
+
+Let's create an array of objects containing simple information about books that will be in this component:
+
+```js
+// bookshelf.js
+export const bookShelf = [
+  {
+    bookName: "Eloquent JavaScript",
+    author: "Marijn Haverbeke",
+  },
+  {
+    bookName: "CSS The Definitive Guide",
+    author: "Eric A. Meyer & Estelle Weyl",
+  },
+  {
+    bookName: "Automate the boring stuff with Python",
+    author: "Al Sweigart",
+  },
+  {
+    bookName: "Make: JavaScript Robotics",
+    author: "Donovan Buck, Bryan Hughes et al",
+  },
+];
+```
+
+The data in `authors.js` will serve as the information that would be mapped onto the `authorsPage` so the search component can be able to filter the list of authors once the user starts typing into the input field.
+
+Let's create the components needed in this app below and I'll explain how everything works.
+
+The `Search` component presents the UI (i.e. It renders an input element where users can filter the books in the list.) to the user. It makes use of React props to pass data down to other components such as the author’s page container component.
+
+```js
+// search.js
+import React from “react”
+
+const Search = ({ val, onSearch, ...props }) => {
+  return (
+    <input
+	  type="text"
+	  value={val}
+	  placeholder="Search for any book"
+	  onChange={onSearch}
+	  {...props}
+	/>
+  );
+};
+
+export default Search
+```
+
+The `Books` component performs almost the same function with the search component. Its only difference has to with the usage of React props to share data amongst the other components in the app.
+
+Take a look at how the `bookShelf` component is being imported into the authors container and how it is being used in the `Books` component.
+
+```js
+// books.js
+const Books = ({ books }) => {
+  return (
+  {books.map((bookShelf, index) => {
+    return (
+	  <ul key={index}>
+	    <li>Book: {bookShelf.bookName}</li>
+		<li>Author: {bookShelf.author}</li>
+	  </ul>
+    )
+  })}
+  )
+}
+export default Books
```

Comment:
Uhm...

That wouldn't be necessary actually, since I'm just passing `bookShelf` as a callback variable/array that comes from the map function.

I've added the import statement of `books` from the `bookShelf.js` file in the appropriate component BTW.